addons/stock/models/stock_move.py

```diff
@@ -225,7 +225,7 @@ def _compute_string_qty_information(self):
     def _check_uom(self):
         moves_error = self.filtered(lambda move: move.product_id.uom_id.category_id.id != move.product_uom.category_id.id)
         if moves_error:
-            user_warning = _('You try to move a product using a UoM that is not compatible with the UoM of the product moved. Please use an UoM in the same UoM category.')
+            user_warning = _('You try to move a product using a UoM (%s) that is not compatible with the UoM (%s) of the product moved (%s). Please use an UoM in the same UoM category.' % (self.product_id.uom_id.name, self.product_uom.name, self.product_id.name))
```

Comment:
Don't use `product_id.name` because if you have variantes, you don't know what is it (and you don't show the default_code). Better to user `product_id.display_name`.

Reply:
Try not modify the original string as breaks the translations in stable version (which we want to avoid).
Instead, try adding new content after the original error message, like a new line with the details.